src/routes/v2/authenticatedSites/collectionPages.ts

```diff
@@ -130,61 +170,84 @@ class CollectionPagesRouter {
           }
         )
       }
+    } else if (newFileName) {
+      updateResp = await this.collectionPageService.rename(
+        userWithSiteSessionData,
+        {
+          oldFileName: pageName,
+          newFileName,
+          collectionName,
+          content: pageBody,
+          frontMatter,
+          sha,
+        }
+      )
     } else {
-      /* eslint-disable no-lonely-if */
-      if (newFileName) {
-        updateResp = await this.collectionPageService.rename(
-          userWithSiteSessionData,
-          {
-            oldFileName: pageName,
-            newFileName,
-            collectionName,
-            content: pageBody,
-            frontMatter,
-            sha,
-          }
-        )
-      } else {
-        updateResp = await this.collectionPageService.update(
-          userWithSiteSessionData,
-          {
-            fileName: pageName,
-            collectionName,
-            content: pageBody,
-            frontMatter,
-            sha,
-          }
-        )
-      }
+      updateResp = await this.collectionPageService.update(
+        userWithSiteSessionData,
+        {
+          fileName: pageName,
+          collectionName,
+          content: pageBody,
+          frontMatter,
+          sha,
+        }
+      )
     }
     res.status(200).json(updateResp)
     return next()
   }
 
-  // Delete page in collection
-  async deleteCollectionPage(req, res, next) {
+  async deleteCollectionPage(req: Request, res: Response, next: NextFunction) {
     const { userWithSiteSessionData } = res.locals
-
     const { pageName, collectionName, subcollectionName } = req.params
     const { error } = DeletePageRequestSchema.validate(req.body)
     if (error) throw new BadRequestError(error.message)
-    const { sha } = req.body
-    if (subcollectionName) {
-      await this.subcollectionPageService.delete(userWithSiteSessionData, {
-        fileName: pageName,
-        collectionName,
+    try {
+      await this.getCollectionPage({
         subcollectionName,
-        sha,
-      })
-    } else {
-      await this.collectionPageService.delete(userWithSiteSessionData, {
-        fileName: pageName,
+        userWithSiteSessionData,
+        pageName,
         collectionName,
-        sha,
       })
+    } catch (_) {
+      res
+        .status(409)
+        .json("The page that you are trying to delete does not exist")
+      return next()
     }
-
-    res.status(200).send("OK")
+    let deleteResp
+    if (subcollectionName) {
+      deleteResp = await this.subcollectionPageService.delete(
+        userWithSiteSessionData,
+        {
+          fileName: pageName,
+          collectionName,
+          subcollectionName,
+          /**
```

Comment:
Note: the sha is later [referenced in Github Service](https://github.com/isomerpages/isomercms-backend/blob/fd30cbe2377c2254aeb3ce3bbdd4ebe72919e95e/src/services/db/GitHubService.js#L208), which computes the sha if it does not exist. Thus, this key should be marked optional